clients/java/src/test/java/io/camunda/client/job/SearchJobTest.java

```diff
@@ -0,0 +1,184 @@
+    assertThat(request.getFilter().getElementInstanceKey()).isNotNull();
+    assertThat(request.getFilter().getElementInstanceKey().get$Eq()).isEqualTo("400");
+    assertThat(request.getFilter().getTenantId()).isNotNull();
+    assertThat(request.getFilter().getTenantId().get$Like()).isEqualTo("tenantId");
+  }
+
+  @Test
+  void shouldSearchJobWithFullSorting() {
+    // when
+    client
+        .newJobSearchRequest()
+        .sort(
+            s ->
+                s.jobKey()
+                    .asc()
+                    .type()
+                    .desc()
+                    .worker()
+                    .asc()
+                    .state()
+                    .desc()
+                    .kind()
+                    .asc()
+                    .listenerEventType()
+                    .desc()
+                    .processDefinitionId()
+                    .asc()
+                    .processDefinitionKey()
+                    .desc()
+                    .processInstanceKey()
+                    .asc()
+                    .elementId()
+                    .desc()
+                    .elementInstanceKey()
+                    .asc()
+                    .tenantId()
+                    .desc())
+        .send()
+        .join();
+
+    // then
+    final JobSearchQueryRequest request =
+        gatewayService.getLastRequest(JobSearchQueryRequest.class);
+    final List<SearchRequestSort> sorts =
+        SearchRequestSortMapper.fromJobSearchQuerySortRequest(
+            Objects.requireNonNull(request.getSort()));
+    assertThat(sorts).hasSize(12);
+    assertSort(sorts.get(0), "jobKey", SortOrderEnum.ASC);
+    assertSort(sorts.get(1), "type", SortOrderEnum.DESC);
+    assertSort(sorts.get(2), "worker", SortOrderEnum.ASC);
+    assertSort(sorts.get(3), "state", SortOrderEnum.DESC);
+    assertSort(sorts.get(4), "kind", SortOrderEnum.ASC);
+    assertSort(sorts.get(5), "listenerEventType", SortOrderEnum.DESC);
+    assertSort(sorts.get(6), "processDefinitionId", SortOrderEnum.ASC);
+    assertSort(sorts.get(7), "processDefinitionKey", SortOrderEnum.DESC);
+    assertSort(sorts.get(8), "processInstanceKey", SortOrderEnum.ASC);
+    assertSort(sorts.get(9), "elementId", SortOrderEnum.DESC);
+    assertSort(sorts.get(10), "elementInstanceKey", SortOrderEnum.ASC);
+    assertSort(sorts.get(11), "tenantId", SortOrderEnum.DESC);
+  }
+
+  private void assertSort(
+      final SearchRequestSort sort, final String name, final SortOrderEnum order) {
+    assertThat(sort.getField()).isEqualTo(name);
+    assertThat(sort.getOrder()).isEqualTo(order);
+  }
+
+  @Test
+  void shouldSearchJobWithFullPagination() {
+    // when
+    client
+        .newJobSearchRequest()
+        .page(
+            p ->
+                p.from(3)
+                    .limit(5)
+                    .searchBefore(Collections.singletonList("b"))
+                    .searchAfter(Collections.singletonList("a")))
```

Comment:
💭 Just FYI, we'll have to adjust this when https://github.com/camunda/camunda/pull/33026 is merged. Or adjust that PR when this is merged first.